### DISCUSSION TITLE
Notes from June 14 Discussion on Sharing Questions

### DISCUSSION THREAD

**Discussion by firasm** [PrairieLearn/PrairieLearn#5907] [Ideas]
I did my best to take some notes about this, look for keywords not completeness or 100% accuracy haha!

# Shared Question Banks

- Syntax (from Carl): In an assessment specify a repo to pull a question from.
    - All of the shared files from that repo get pulled in
    - May want to restructure how repos are done, have a separate course for questions and other course stuff	
	- Permissions and access need to be considered
	- Troubles with questions get filed in the repo where the question originated

- Current level of thinking (Matt)

	- Ignoring atribution and permission stuff (which is critical)
	- Fundamental idea: in an assessment specify how to access a question using some type of syntax: @CS374/questions/question_to_be_pulled
	- On the backend, this part is not complicated
	- Complexity comes in permissions, issues, versioning, search, discoverability

- [Nathan's RFC doc](https://github.com/PrairieLearn/PrairieLearn/blob/master/rfcs/001-shareable-courses.md)
	- Question: should we allow for sharing between PrairieLearn instances?
	- Answer: Yes, it should be implemented with sharing between instances in mind

- Recap for Jeff + discussion about clarifying on what a link to a repo would look like

- Cinda brought up how to 

- Nathan brought up federation of different PL instances
	- Kicked down the road

- Dave brought up having different access requirements (view, edit, copy) of a question.

- Requirements from Carl:
	- Discoverability not important at the moment,
	- Multiple repos to pull questions from
	- Separation of upper level students as staff in lower-year CS courses
	- Repos grant other courses access
	- Need to preserve the issue reporting: when a student reports an error, how does it get viewed from a course instance
	- Ideally: push student report, but without student information
	- Some discussion about who deals with the reports/issues

- Comment from Jeff: Should not assume there is someone else on the other end that is able to fix the question. Should be able to co

- Comment from Cinda: Attribution is important for sharing questions with folks
	- Matt: Automatically generate reports on "who is using this question, or my questions"
	- Tags and dates is a useful amount of metadata

- Comment from Nathan: Berkeley was considering adding a README.md file in every question repo

- Comment about setting Prairielearn as an "app" that inherits permissions from an access token
	- Don't want to lock-in to GitHub.com, want to support multi-institution which have GHEs, BitBucket, GitLab, etc...

- Question about how to hide students 

- Different cases for access/permissions:
	- Staff members have view access to allow students to view their tests, will do no development; may be taking CS 374, should not see questions that are used in CS 225
	- Staff members that are currently taking 374, need edit access to develop course CS 225
	- Staff members need access to everything, 
	- Matt said this possibly already works
	- Conclusion: new design must have support for pulling questions from repositories

**Comment by echuber2** [comment]
Related: "Bring your own repo" discussion https://github.com/PrairieLearn/PrairieLearn/discussions/6144
